index.js

```diff
@@ -120,34 +69,25 @@ module.exports = (str, opts) => {
 		}
 	}
 
-	// Remove URL with empty query string
-	if (urlObj.search === '?') {
-		delete urlObj.search;
-	}
-
-	const queryParameters = queryString.parse(urlObj.search);
-
 	// Remove query unwanted parameters
 	if (Array.isArray(opts.removeQueryParameters)) {
-		for (const key in queryParameters) {
-			if (testParameter(key, opts.removeQueryParameters)) {
-				delete queryParameters[key];
-			}
-		}
+		[...urlObj.searchParams.keys()]
+			.reduce((acc, key) => {
```

Comment:
Why complicate it with `reduce`? It could simply be:

```js
// Remove query unwanted parameters
if (Array.isArray(opts.removeQueryParameters)) {
	[...urlObj.searchParams.keys()]
		.filter(key => testParameter(key, opts.removeQueryParameters))
		.forEach(key => urlObj.searchParams.delete(key));
}
```

Reply:
Actually, even better:

```js
// Remove query unwanted parameters
if (Array.isArray(opts.removeQueryParameters)) {
	for (const key of [...urlObj.searchParams.keys()]) {
		if (testParameter(key, opts.removeQueryParameters)) {
			urlObj.searchParams.delete(key);
		}
	}
}
```

Reply:
updated, thanks!